zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationProvider.java

```diff
@@ -196,11 +197,11 @@ public String getScheme() {
      */
     protected String getClientId(X509Certificate clientCert) {
         String clientCertIdType =
-            System.getProperty(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENTCERTIDTYPE);
+            System.getProperty(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENT_CERT_ID_TYPE);
         if (clientCertIdType != null && clientCertIdType.equalsIgnoreCase("SAN")) {
             try {
                 return matchAndExtractSAN(clientCert);
-            } catch (CertificateException | IllegalArgumentException ce) {
+            } catch (Exception ce) {
```

Comment:
Just curious, why do we need to catch all Exceptions here? This means we will always warning instead of stopping the call even some unexpected error happens.

Reply:
The reason we have a try-catch clause here instead of throwing an Exception is to allow the authentication logic to proceed. The code states that if there are any errors, we want to resort to using Subject Domain Name by default, and that is what this does, regardless of what kind of runtime exceptions may occur.